### ISSUE TITLE
Add option to exclude components

### ISSUE THREAD

**Issue by epicEaston197** [Nytra/ResoniteDistantComponentOptimizer#4]
I want an option so I can exclude components like Panner1D and particle system from the optimizer Especially the particle system as it's already asynchronous and already slows down its update rate so it doesn't really make sense to slow down the update rate for that component

**Comment by Nytra** [comment]
ParticleSystem isn't affected by this mod

**Comment by epicEaston197** [comment]
> ParticleSystem isn't affected by this mod

okay?
that was just one of the examples It would still be nice to be able to exclude components

**Comment by Nytra** [comment]
Yes I can add that, I already have it working in a test build

**Comment by Nytra** [comment]
On further investigation if I add Exclude Types it actually makes the FPS worse, so I may need to reconsider how to do this without making the FPS worse

**Comment by Banane9** [comment]
Makes sense, checking even a hashset for hundreds if not thousands of components' types is a lot of overhead to be doing every frame.

**Comment by Nytra** [comment]
I may need to switch to patching every component type manually instead of patching the base type, then I can just skip patching ones which should be excluded

**Comment by Banane9** [comment]
That won't work, at least not without finding something else to patch - InternalRunUpdate doesn't exist on anything else.
Same for the CanRunUpdates property I've tried using only existing on Component.

With pre-patching, it would be possible to make it overridable and add implementations of methods though.

**Comment by Nytra** [comment]
Well I was thinking I could just patch OnCommonUpdate

**Comment by Banane9** [comment]
Possible, although many components also don't have that - though I guess those aren't the ones causing much performance overhead lol
